addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1586,8 +1586,13 @@ export class PosStore extends Reactive {
         if (name === "ProductScreen") {
             this.get_order()?.deselect_orderline();
         }
-        this.previousScreen = this.mainScreen.component?.name;
         const component = registry.category("pos_screens").get(name);
+        if (
+            (component.updatePreviousScreen ?? true) &&
+            (this.mainScreen.component?.updatePreviousScreen ?? true)
+        ) {
```

Comment:
```suggestion
        if (!this.mainScreen.component?.disablePreviousScreen) {
```

I don't really understand why its necessary to check if the next component can update the previous screen. 

Just check if the current one can update it. By doing that we ensure that "ScreenSaver" will never be the previousScreen.

Reply:
@davidmonnom We lose the correct previous screen when the screen saver is displayed. For example, if you are on screen A and the previous screen was B, your suggestion would update the previous screen to A when the screen saver appears. I implemented it so that screens we don’t want to record as the previous screen won’t change the original previous screen when displayed.

Reply:
When the screenSaver appears, we want to return to the screen that was displayed just before, for example:
- ProductScreen
- TicketScreen
- ScreenSaver
- TicketScreen